docs/developers_notes/03-base_regressor.md

```diff
@@ -0,0 +1,59 @@
+# The Abstract Class `BaseRegressor`
+
+`BaseRegressor` is an abstract class that inherits from `Base`, stipulating the implementation of number of abstract methods such as `fit`, `predict`, `score`. This ensures seamless assimilation with `scikit-learn` pipelines and cross-validation procedures.
+
+
+!!! Example
+    The current package version includes a concrete class named `nemos.glm.GLM`. This class inherits from `BaseRegressor`, which in turn inherits `Base`, since it falls under the " GLM regression" category. 
+    As any `BaseRegressor`, it **must** implement the `fit`, `score`, `predict` methods.
```

Comment:
I don't understand this sentence. I think what you mean is either "As a `BaseRegressor`, it must implement..." or "Like any `BaseRegressor`, it implements..."